rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,501 @@
+- (-) CommonMark does not specify the current rich features from nixdoc, such as predefined sections and structures that could be used as a source for automated toolings. Such as types for type-checking or examples to run automated-tests
+    - (+) Future tools can still build their conventions on top of this RFC. They might not directly specify them in an RFC but be a tool developers choose for a specific codebase. However, we have yet to get that tools. So it is good when rich features remain unspecified.    
+  
+</details>
+
+## Place doc-comments before the referenced expression 
+
+**Observing**: Doc-comments currently are placed above the expression they document. More precisely, only named attribute bindings `foo = <expr>` can be documented. There is also the need to support documentation for anonymous functions. More generally, there is the need to document anonymous expressions where they are defined.
+
+**Considering**: General reference logic between doc-comments and expressions.
+
+**Decision**: Doc-comments refer to the **immediately** preceded expression. Only whitespaces are allowed in between.
+
+> Note: We are working on a generic example implementation that covers about 90% of all documentation use cases.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) Doc-comments should have only one variant to reduce complexity. Referencing the next node seems straightforward. 
+- (-) A variant that references the previous node in the AST should be avoided to reduce complexity.
+- (+) Relation between documentation and the referenced implementation is straightforward and back-trackable.
+- (-) Concrete Implementation might be complex.
+- (-) Unclear if complete documentation might also need backward references. 
+    - (-), e.g., rust uses backward references only at the top of file comments.
+    - (+) Nix files can only have ONE expression; the next AST Node, in case of top-of-file comments, is thus always only that one expression. (Unlike in Rust)
+- (-) Tools need to be smart enough to understand asignments `=` and other forms of creating names for anonymous expressions. (e.g., `callPackage` and `import` )
+    - (+) Tools can still come up with other solutions that do not involve calculating everything dynamically from nix code but could also involve a static configuration.
+    - (+) The whole `tool point` is an implementation detail as long as it is not impossible. The current tool, `nixdoc`, already proves that it is possible to have static documentation to a certain degree.
+
+</details>
+
+## Single-line doc-comments (do not exist)
+
+**Observing**: Nix offers two variants of comments; single- (`#`) and multi-line comments (`/* */`). There may be use cases where it is desirable to have a form of single-line comments subtyped for doc-comment purposes. 
+
+**Considering**: Single-line comment for documentation.
+
+**Decision**: Single-line comments **cannot be used** in any form for documentation puposes.
```

Comment:
This should probably clarify you're talking about the comment syntax of single-line comments, not about the contents themselves having to be more than one line. The `/** {content} */` is a single line.

Reply:
Yes good point. I am talking about the „#“ variant of a comment. 
I should clarify it better